packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -4911,7 +4913,7 @@ class _UpdateTextSelectionAction<T extends DirectionalCaretMovementIntent> exten
 
     final bool shouldCollapseToBase = intent.collapseAtReversal
       && (selection.baseOffset - selection.extentOffset) * (selection.baseOffset - newSelection.extentOffset) < 0;
-    final TextSelection newRange = shouldCollapseToBase ? TextSelection.fromPosition(selection.base) : newSelection;
```

Comment:
Could you explain this change? We do care about the affinity here I think?

Reply:
I made this change because when  `selection.base` had an `upstream` affinity, for example when we have a word wrapped line and place the caret in the `downstream` position at the word wrap and try to get the leading paragraph boundary, the `selection.extendTo` will create a selection with an `upstream` affinity, so when trying to collapse the selection after, the caret is placed at `upstream` position at the word wrap. It is expected that it would be in the `downstream` position of the word wrap where the selection began. 

I can also confirm this behavior on native macOS. And from my testing the change hasn't affected other shortcuts.

Reply:
Is that the case for word boundary too? If it is could you add a test to make sure this behavior won't get changed?

Reply:
Offline: word boundary on native macOS also behaves this way (defaults to `downstream`), but native Linux and Windows seem to match the behavior where they care about the `affinity`. I'm going to hold off on this `.fromPosition` change since it functionally doesn't break anything, but I will open up an issue to fix this default behavior on macos.